dowhy/causal_model.py

```diff
@@ -283,6 +283,10 @@ def estimate_effect(self, identified_estimand, method_name=None,
                                   treatment_value=treatment_value)
         else:
             if fit_estimator:
+                if method_params is not None and num_components <= 2:
```

Comment:
`num_components <=2` is not a robust check for dowhy internal. Because a user can also provide ``backdoor.dowhy.propensity_score_stratification" and the API is constructed to handle that, in addition to "backdoor.propensity_score_stratification". (I notice that the inline comment on line 255 is not accurate) Maybe you can check `num_components <=2 or method_name.contains("dowhy")`.

Reply:
Would this work `(num_components <= 2 or estimator_package=='dowhy')` ? 
Is a bit more defensive, to prevent triggering on things like `other_package.dowhy.estimator`